src/transaction/mod.rs

```diff
@@ -0,0 +1,102 @@
+use ckb_types::packed::Script;
+
+use crate::{tx_builder::TxBuilderError, NetworkInfo};
+
+use self::{builder::FeeCalculator, handler::ScriptHandler};
+
+pub mod builder;
+pub mod handler;
+pub mod input;
+pub mod signer;
+
+pub struct TransactionBuilderConfiguration {
+    pub network: NetworkInfo,
+    pub script_handlers: Vec<Box<dyn ScriptHandler>>,
+    pub fee_rate: u64,
+    pub small_change_action: SmallChangeAction,
+}
+
+/// Define what to do when change capacity is to small to create a new cell.
+/// If the change capacity is lower than `threshold` in shannons, it is small.
+pub enum SmallChangeAction {
+    /// Find another input cell, and put it's capacity to change.
+    /// It's the default action.
+    FindMoreInput,
+    /// Put the change capacity to the first output cell with target address.
+    /// If change capacity lower than threshold, add it to output cell,
+    /// or it will act as `FindMoreInput`
+    ToOutput { target: Script, threshold: u64 },
+    /// Put the small change capacity to fee.
+    /// If change capacity lower than threshold, add it to fee, or it will act as `FindMoreInput`.
+    /// *Note*:
+    /// If the threshold is 61CKB (6100000000 shannons) and assume the mimimum capacity for a cell is 61 bytes,
+    /// the transaction fee might bigger than 61 CKBs, because to create a change cell,
+    /// will need extra transaction fee for the change cell.
```

Comment:
```suggestion
    /// the transaction fee might be higher than 61 CKBs, because it will need an extra transaction fee to create a change cell.
```

Reply:
The note part should be removed, because in previous version, add a change cell first, then do the calculation, if change lower than minimum cell capacity, remove the change call, and original change + released fee for change cell may higher than minimum cell.

Now only check before a change cell created, after created, it will not be removed, so it won't exceed the threshold.